### ISSUE TITLE
Unable to select authorized account on page load

### ISSUE THREAD

**Issue by acegikmedia** [google/google-api-javascript-client#361]
When multiple accounts are signed in, it always prompts the user to sign in on each page where we're pulling data from Google.  I can't seem to find any way to specify the account to use.  Is there a way to pass their GoogleID or email when checking if signed in?

**Comment by TMSCH** [comment]
Could you explain with a bit more details your problem? What do you mean by prompt the user? Are you using a JS library? Could you provide a snippet to reproduce the issue?

**Comment by acegikmedia** [comment]
So using the quickstart found here, when multiple users are signed in, gapi.auth2.getAuthInstance().isSignedIn.get() always returns false and forces the user to select the account they want to proceed with.  This is fine for the initial sign-in/authorization, but it then displays everytime the page is refreshed.  Is there a way to pass the user credentials or access token to avoid the redundant authorization button from appearing each time?

`// On load, called to load the auth2 library and API client library.
function handleClientLoad() {
	gapi.load('client:auth2', initClient);
}

// Initializes the API client library and sets up sign-in state listeners.
function initClient() {
	gapi.client.init({
		discoveryDocs: DISCOVERY_DOCS,
		clientId: CLIENT_ID,
		scope: SCOPES
	}).then(function () {
		// Listen for sign-in state changes.
		gapi.auth2.getAuthInstance().isSignedIn.listen(updateSigninStatus);

  		// Handle the initial sign-in state.
  		updateSigninStatus(gapi.auth2.getAuthInstance().isSignedIn.get());
  		authorizeButton.onclick = handleAuthClick;
	});
}

// Called when the signed in status changes, to update the UI appropriately. After a sign-in, the API is called.
function updateSigninStatus(isSignedIn) {
	if (isSignedIn) {
		authorizeButton.style.display = 'none';
		listUsers();
	} else {
		authorizeButton.style.display = 'block';
	}
}

// Sign in the user upon button click.
function handleAuthClick(event) {
	gapi.auth2.getAuthInstance().signIn();
}

function appendPre(message) {
	var pre = document.getElementById('content');
	var textContent = document.createTextNode(message + '\n');
	pre.appendChild(textContent);
}`

**Comment by TMSCH** [comment]
@acegikmedia this sounds like a bug, the user should remain signed in. What browser/version/platform are you using?

**Comment by acegikmedia** [comment]
@TMSCH I've tested this on Windows 10 in both Chrome (62.0.3202.94) and Edge (40.15063.674.0).  Only tested so far on localhost.

**Comment by TMSCH** [comment]
Just to confirm: first, you sign in with one of your Google account currently logged in. It successfully signs in the user. Then, you refresh the page. At this point, instead of having the user signed in, it does not. Is that correct?

If not, please write the step by step instructions to reproduce the error.

**Comment by acegikmedia** [comment]
Correct.  User will initially sign with their Google account, which works.  Then upon page refresh or navigating to another page within the site, it returns false for isSignedIn, prompting the user to sign in again.

Everything works perfectly when only one Google account is signed in, but fails if the user has multiple accounts signed in.

**Comment by TMSCH** [comment]
@acegikmedia do you block third-party cookies? Are you using InPrivate mode?

**Comment by acegikmedia** [comment]
No I’ve made sure cookies are not blocked.. and no inprivate mode is not used.

**Comment by andriychuma** [comment]
It looks like I faced with the same issue. When I clean up Google sign in data in Chrome or use incognito mode, then sign in to Google via my app, everything works well. User stays signed in after page refresh.

But when I log in with a second account (so that Chrome stores two logged in users), everything works until page refresh. When I refresh the page, after the `client.init(...)` call the user is considered signed out (`gapi.auth2.getAuthInstance().isSignedIn.get() === false`).

Since the client library source code is still not available I needed to investigate the minified file. Here's what I found and what probably could help us to find the root cause:
1. Negative use case happens on `localhost` without a port. When there is a port provided like `localhost:4200`, everything works well.
2. Negative use case has two requests to `https://accounts.google.com/o/oauth2/iframerpc` with `action=checkOrigin` and `action=listSessions` query params. Whilst the positive one contains `action=checkOrigin` and `action=issueToken`.
3. There's a method that likely doesn't correctly handle `http://localhost` input:
```
pe.prototype.Ua = function(a) {
    var b = [];
    if (a) {
        var c = document.createElement("a");
        c.href = a;
        if ("https:" == c.protocol && ("" == c.port || 0 == c.port || 443 == c.port) || "http:" == c.protocol && ("" == c.port || 0 == c.port || 80 == c.port))
            for (a = c.hostname.split("."); 1 < a.length; )
                b.push(c.protocol + "//" + a.join(".")),
                "https:" == c.protocol && b.push("http://" + a.join(".")),
                a.shift();
        else
            b.push(a)
    }
    return b
}
```
Because of `1 < a.length` condition it returns empty array `[]` instead of `['http://localhost']`.

**Comment by dvanderb** [comment]
This looks like a question about Google Login and Authentication. Please address this question to the google-signin tag on stackoverflow(https://stackoverflow.com/questions/tagged/google-signin)

If you believe this is in error, please reopen the issue.
